plugins/filter/merge_data.py

```diff
@@ -0,0 +1,335 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2022     , DEMAREST Maxime <maxime@indelog.fr>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+name: merge_data
+short_description: Merge dict or lists by ensuring the comparing datas are
+                   presents, absents or identics.
+version_added: 5.0.2
+author: DEMAREST Maxime (@indelog)
+description:
+  - Merge two datas, current and expected in form of dict or list.
+  - You can have mixed data like dict->list or list->dict. If the two type
+    are different, the expected datas are always returned.
```

Comment:
I don't think I understand this. Do you mean that one argument is a dictionary, while the other a list? And with "expected", do you mean the input named "expected" (should we written as `I(expected)`), or the english word "expected"?